src/pages/Recovery/Recovery.tsx

```diff
@@ -0,0 +1,223 @@
+    [KnownChainIds.OsmosisMainnet]: [OsmosisChainAdapter.defaultBIP44Params],
+    [KnownChainIds.BitcoinMainnet]: [
+      { ...BitcoinChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...BitcoinChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...BitcoinChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+    [KnownChainIds.BitcoinCashMainnet]: [
+      { ...BitcoinCashChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...BitcoinCashChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...BitcoinCashChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+    [KnownChainIds.LitecoinMainnet]: [
+      { ...LitecoinChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...LitecoinChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...LitecoinChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+    [KnownChainIds.DogecoinMainnet]: [
+      { ...DogecoinChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...DogecoinChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...DogecoinChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+  }
+}
+
+type GetDangerousAddressesArgs = {
+  wallet: HDWallet
+}
+
+type GetDangerousAddresses = (
+  args: GetDangerousAddressesArgs,
+) => Promise<GetDangerousAddressesReturn>
+type GetDangerousAddressesReturn = {
+  input: {
+    targetChainId: ChainId
+    chainName: string
+    bip44Params: BIP44Params
+    accountType: UtxoAccountType | undefined
+  }
+  output: {
+    pubkey: string
+    addresses: string[]
+  }
+}[]
+
+const getDangerousAddresses: GetDangerousAddresses = async ({ wallet }) => {
+  const dangerousBip44ParamsByChainId = getDangerousBIP44ParamsByChainId()
+
+  if (!supportsBTC(wallet)) {
+    throw new Error(`Recovery: wallet does not support bitcoin`)
+  }
+
+  const chainAdapterManager = getChainAdapterManager()
+
+  const result = []
+
+  for (const affectedChainId of affectedChainsIds) {
+    const maybeChainAdapter = chainAdapterManager.get(affectedChainId)
+    if (!maybeChainAdapter) throw new Error(`Recovery: no chain adapter for ${affectedChainId}`)
+    const adapter = maybeChainAdapter as unknown as UtxoBaseAdapter<UtxoChainId>
+    const chainName = adapter.getName()
+    for (const [bip44ParamChainName, wrongChainId] of Object.entries(KnownChainIds)) {
+      const bip44ParamsArray = dangerousBip44ParamsByChainId[wrongChainId]
+      for (const bip44Params of bip44ParamsArray) {
+        // only segwit native produces bc1-prefixed addresses
+        const accountType = UtxoAccountType.SegwitNative
+        const publicKey = await getDangerousPublicKey(wallet, bip44Params, accountType)
+        const addresses: string[] = []
+        // generate addresses
+        for (let index = 0; index < addressCount; index++) {
+          // only segwit native produces bc1-prefixed addresses
+          const scriptType = BTCInputScriptType.SpendWitness // segwit-native
+          const address = await wallet.btcGetAddress({
```

Comment:
note for @kaladinlight so i don't forget, i think this is wrong/needs to be done at a lower level